### PR TITLE
print usage required during usage validation

### LINKED ISSUE #339: usage panic needs better error message
Currently I'm debugging why my previously working old cold now fails with an assert for usage, which prints:
```
thread 'main' panicked at 'assertion failed: buffer.usage.contains(usage)', /Users/ataraxis/.cargo/git/checkouts/wgpu-53e70f8674b08dd4/c3609d7/wgpu-native/src/device.rs:1218:21
```

This is pretty unhelpful to users; if we're doing proper validation that users can use to fix their code, it should at least print which usage it is expecting to be present (and hence triggering the assert)

### PR DESCRIPTION
**Pull request by m4b**
fixes #339

### CODE REVIEW

**Comment by m4b** [comment]
i used the web editor because i'm lazy, don't judge me

File: wgpu-native/src/device.rs [hunk 1]
```diff
@@ -1215,7 +1215,7 @@ pub fn device_create_bind_group<B: GfxBackend>(
                         .buffers
                         .use_extend(&*buffer_guard, bb.buffer, (), usage)
                         .unwrap();
-                    assert!(buffer.usage.contains(usage));
+                    assert!(buffer.usage.contains(usage), format!("Invalid usage for buffer {:?}, {:?} is required", buffer.usage, usage));
```
Comment (kvark) [inline wgpu-native/src/device.rs]: first of all, `format!` isn't needed when you are in the `assert!` macro
Secondly, we don't really need to print the `buffer.usage` here. There is no other way to fix it than going to the code responsible for buffer creation and checking/fixing the usage flags, so it's not like we'd be providing any info that the user wouldn't see anyway. We only need to print the `usage` here.

Comment (m4b) [reply wgpu-native/src/device.rs]: How does the user know which buffer caused this ? I didn’t. Having which usages were passed was very helpful. I always think it’s better to give as much possible information during a crash as you can, so why not pass it ?

**Comment by m4b** [comment]
Ok i just needed to force push 3 times, i think this is more in line with what you wanted ;)

**Comment by kvark** [comment]
bors retry

**Pull request merged**